src/components/user-details/user-details.tsx

```diff
@@ -0,0 +1,58 @@
+import { useNavigate } from 'react-router-dom';
+import { fetchUser } from '../../api/fetch-user';
+import { H1, Paragraph } from '../../styles/text-styles';
+import { ErrorMessage } from '../forms/error-message';
+import { LoadingIndicator } from '../loading/loading-indicador';
+import { Title } from '../user-list/user-list-styles';
+import { StrongText, UserDetailsContainer, UserInfo } from './user-details-styles';
+
+interface UserDetailsProps {
+  userId: string;
+  token: string;
+}
+
+export const UserDetails = ({ userId, token }: UserDetailsProps) => {
```

Comment:
Se a resposta da pergunta anterior for que ele quebra e não mostra um erro poderia deixar o userId aqui como opcional e verificar nesse componente se o param veio ou não, se veio vc mostra o componente e se não vc mostra um erro de feedback para o usuário falando o que aconteceu.